addon/-private/diff-route-info.js

```diff
@@ -0,0 +1,118 @@
+import { assign } from '@ember/polyfills';
+import { gte } from 'ember-compatibility-helpers';
+
+export let qpsChanged;
+export let shouldRefreshModel;
+export let pathsDiffer;
+export let paramsDiffer;
+export let createPrefetchChangeSet;
+
+if (gte('3.6.0')) {
+  function createList(enumerable) { // eslint-disable-line no-inner-declarations
+    let out = [];
+
+    if (enumerable === null) return out;
+
+    enumerable.find(item => {
```

Comment:
Reading this, I was surprised/confused, after digging in I realized:

* `enumerable` is a RouteInfo or RouteInfoWithAttributes or null, not really an enumerable.
* `routeInfo.find` with implicit undefined return value happens to iterate the full list

To aide readers like me, could I recommend:

* explicit `return false;` + short comment `// using find` to enumelate forEach by always returning false` in the find callback?


Also, I believe if this was TS it would error as they expect the predicate to find to return a Boolean: https://github.com/tildeio/router.js/blob/f89a80f09bac16d6d8cea32a5404fb8bf2adf3d1/lib/router/route-info.ts#L49